kernel/aster-nix/src/arch/x86/exception.rs

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use ostd::cpu::*;
+
+use crate::prelude::*;
+
+macro_rules! log_trap_common {
+    ($exception_name: ident, $trap_info: ident) => {
+        trace!(
+            "[Trap][{}][err = {}]",
+            stringify!($exception_name),
+            $trap_info.error_code
+        )
+    };
+}
+
+pub fn log_trap_info(exception: &CpuException, trap_info: &CpuExceptionInfo) {
```

Comment:
This logging utility function does not have to be architecture dependent. What's really architecture dependent is the type definition of `CpuException`. You can modify the `define_cpu_exception` macro in `ostd` so that the `Debug` implementation of `CpuException` can pretty print the exception names. To do so, I think the `number` field of `CpuException` should be changed from an integer to an enum.

```rust
pub struct CpuException {
    id: CpuExceptionId, // I think ID is better than number
    typ: CpuExceptionType,
}

#[derive(Debug)]
pub enum CpuExceptionId {
    DIVIDE_BY_ZERO = 0,
    // ..
}
```

Using enum and derive, the value `CpuExceptionId::DIVIDE_BY_ZERO` will be automatically pretty printed as `DIVIDE_BY_ZERO`, thus freeing us from doing so manually (as done in the current `log_trap_info`).

Reply:
Another thing is that, currently `log_trap_info` will print the page fault's address if it is. So if we want to keep this information, maybe we should merge #1087 first, which provides `.is_page_fault()` and `.page_fault_address()`?

Reply:
Sure. We can go with #1087 first.